Adds End-Use Subcategory field for Pumps

shorowit opened the pull request:
Pull request overview
---------------------
Adds an End-Use Subcategory field to all `Pump` objects (similar to `Fan` objects). Note that the Pumps table header has been named "End Use Subcategory" for consistency with the pending bugfix for [this issue](https://github.com/NREL/EnergyPlus/issues/6949).

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check
 - [x] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [x] Verify naming conventions and styles, memos and notes and defaults
   - [x] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [x] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [x] If output changes, including tabular output structure, add to output rules file for interfaces

shorowit committed:
Initial commit to add end use subcategory field for pumps.

shorowit committed:
Add new field to a few test IDFs.

shorowit committed:
Bugfix.

shorowit committed:
Add unit tests for each pump type.

nrel-bot-3 commented:
@shorowit @lgentile it has been 28 days since this pull request was last updated.

mjwitte commented on src/EnergyPlus/OutputReportPredefined.cc:
```diff
@@ -1885,6 +1887,7 @@ namespace OutputReportPredefined {
         pdchPumpFlow = newPreDefColumn(pdstPump, "Water Flow [m3/s]");
         pdchPumpPower = newPreDefColumn(pdstPump, "Electric Power [W]");
         pdchPumpPwrPerFlow = newPreDefColumn(pdstPump, "Power Per Water Flow Rate [W-s/m3]");
+        pdchPumpEndUse = newPreDefColumn(pdstPump, "End Use Subcategory");
```
Was there a reason for putting the subcategory before the motor efficiency in the table?  Looks a little odd to me to separate the motor efficiency from the other power-related columns.

shorowit replied:
OK, I moved it to the last column.

mjwitte reviewed (commented):
@shorowit This looks good. The I/O Ref and idd changes look good. Ran a couple of test files locally, testing random strings in the new subcategory field. 

One more thing is needed, a new file: /src/Transition/OutputRulesFiles/OutputChanges9-0-0-to-9-1-0.md that describes the new column in the pumps table. Copy and edit [OutputChanges8-9-0-to-9-0-0.md](https://github.com/NREL/EnergyPlus/blob/develop/src/Transition/OutputRulesFiles/OutputChanges8-9-0-to-9-0-0.md). Also, please mention the heading name change for Fans that was done in #7068.
And please merge up to current develop. I'll merge after CI comes back green.

shorowit committed:
Change table column order.

shorowit committed:
Adds OutputChanges9-0-0-to-9-0-1.md

shorowit committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into pump_subcategories

shorowit commented:
@mjwitte I believe I addressed your comments. Is anything needed for input transition? The new field was added at the end of each object, so I assume not. If that's right, then I think this is ready to go.

mjwitte commented:
@shorowit No transition required. Thanks for adding the output changes doc, except it should be named "OutputChanges9-0-0-to-9-1-0.md" (not 9-0-1).

shorowit committed:
Fix filename.

mjwitte reviewed (approved):
CI is all green. Will merge after problem merging #7033 is resolved.
